samples/client/petstore-security-test/typescript-angular/api/fake.service.ts

```diff
@@ -57,51 +59,38 @@ export class FakeService {
 
     /**
      * To test code injection *_/ &#39; \&quot; &#x3D;end -- \\r\\n \\n \\r
-     * 
-     * @param testCodeInjectEndRnNR To test code injection *_/ &#39; \&quot; &#x3D;end -- \\r\\n \\n \\r
+     * To test code injection *_/ &#39; \&quot; &#x3D;end -- \\r\\n \\n \\r
+     * @param UNKNOWN_BASE_TYPE 
      * @param observe set whether or not to return the data Observable as the body, response or events. defaults to returning the body.
      * @param reportProgress flag to report request and response progress.
      */
-    public testCodeInjectEndRnNR(testCodeInjectEndRnNR?: string, observe?: 'body', reportProgress?: boolean): Observable<any>;
-    public testCodeInjectEndRnNR(testCodeInjectEndRnNR?: string, observe?: 'response', reportProgress?: boolean): Observable<HttpResponse<any>>;
-    public testCodeInjectEndRnNR(testCodeInjectEndRnNR?: string, observe?: 'events', reportProgress?: boolean): Observable<HttpEvent<any>>;
-    public testCodeInjectEndRnNR(testCodeInjectEndRnNR?: string, observe: any = 'body', reportProgress: boolean = false ): Observable<any> {
+    public testCodeInjectEndRnNR(UNKNOWN_BASE_TYPE?: UNKNOWN_BASE_TYPE, observe?: 'body', reportProgress?: boolean): Observable<any>;
```

Comment:
`./bin/security/typescript-angular.sh` had warned:
```
[main] WARN  o.o.codegen.DefaultCodegen - The following schema has undefined (null) baseType. It could be due to form parameter defined in OpenAPI v2 spec with incorrect consumes. A correct 'consumes' for form parameters should be 'application/x-www-form-urlencoded' or 'mul
tipart/form-data' 
```
And I changed modules/openapi-generator/src/test/resources/2_0/petstore-security-test.yaml:
```diff
       consumes:
-        - application/json
-        - "*/ ' \" =end -- \r\n \n \r"
+        - application/x-www-form-urlencoded
```

But invalid identifier was generated:
```diff
-    public testCodeInjectEndRnNR(UNKNOWN_BASE_TYPE?: UNKNOWN_BASE_TYPE, observe?: 'body', reportProgress?: boolean): Observable<any>;
-    public testCodeInjectEndRnNR(UNKNOWN_BASE_TYPE?: UNKNOWN_BASE_TYPE, observe?: 'response', reportProgress?: boolean): Observable<HttpResponse<any>>;
-    public testCodeInjectEndRnNR(UNKNOWN_BASE_TYPE?: UNKNOWN_BASE_TYPE, observe?: 'events', reportProgress?: boolean): Observable<HttpEvent<any>>;
-    public testCodeInjectEndRnNR(UNKNOWN_BASE_TYPE?: UNKNOWN_BASE_TYPE, observe: any = 'body', reportProgress: boolean = false ): Observable<any> {
+    public testCodeInjectEndRnNR(testCodeInject*#39;&quot;&#x3D;endRnNR?: string, observe?: 'body', reportProgress?: boolean): Observable<any>;
+    public testCodeInjectEndRnNR(testCodeInject*&#39;&quot;&#x3D;endRnNR?: string, observe?: 'response', reportProgress?: boolean): Observable<HttpResponse<any>>;
+    public testCodeInjectEndRnNR(testCodeInject*&#39;&quot;&#x3D;endRnNR?: string, observe?: 'events', reportProgress?: boolean): Observable<HttpEvent<any>>;
+    public testCodeInjectEndRnNR(testCodeInject*&#39;&quot;&#x3D;endRnNR?: string, observe: any = 'body', reportProgress: boolean = false ): Observable<any> {
```

Reply:
I think [AbstractTypeScriptClientCodegen#toParamName](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java#L162) is broken:

```diff
-        name = sanitizeName(name, "\\W-[\\$]");
+        name = sanitizeName(name, "[^\\w$]");
```

Reply:
@grktsh can you include this fix, too?

Reply:
@macjohnny I will!

Reply:
@macjohnny Done.  Could you review again?

Reply:
`"*/ ' \" =end -- \r\n \n \r"` is for detecting code injection (malicious input in `consumes`). I'll add it back with a separate PR.

Reply:
Done via d57009d14d